### PR TITLE
[New Integration] TAXII feed

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
Ready

## Related Issues
requested in: https://github.com/demisto/etc/issues/19714

## Description
Implemented TAXII Feed Integration - fetch indicators from TAXII server.

## Required version of Demisto
5.5.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical Writer Review @kirbles19:
- [ ] [RN](https://github.com/demisto/content/edit/taxii-feed/Integrations/FeedTAXII/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5268)
- [ ] [YAML](https://github.com/demisto/content/edit/taxii-feed/Integrations/FeedTAXII/FeedTAXII.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F5268)
- [ ] [Detailed description](https://github.com/demisto/content/edit/taxii-feed/Integrations/FeedTAXII/FeedTAXII_description.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5268)

### CODE REVIEW

File: Integrations/FeedTAXII/FeedTAXII.py [hunk 1]
```diff
@@ -0,0 +1,918 @@
...
+            LOG('Unknown AddressObjectType category: {!r}'.format(acategory))
+            return []
+
+        return [{
+            'indicator': indicator,
+            'type': type_
+        }]
+
+
+''' domainnameobject.py '''
```
Comment (yuvalbenshalom) [inline Integrations/FeedTAXII/FeedTAXII.py]: remove

File: Integrations/FeedTAXII/FeedTAXII.py [hunk 2]
```diff
@@ -0,0 +1,918 @@
...
+        if domain is None:
+            return []
+
+        return [{
+            'indicator': domain.string.encode('ascii', 'replace').decode(),
+            'type': 'Domain'
+        }]
+
+
+''' fileobject.py '''
```
Comment (yuvalbenshalom) [inline Integrations/FeedTAXII/FeedTAXII.py]: remove

File: Integrations/FeedTAXII/FeedTAXII.py [hunk 3]
```diff
@@ -0,0 +1,918 @@
...
+# disable insecure warnings
+urllib3.disable_warnings()
+
+EPOCH = datetime.utcfromtimestamp(0).replace(tzinfo=pytz.UTC)
+INTEGRATION_NAME = 'TAXII1'
+
+''' addressobject.py '''
+
+
+class AddressObject(object):
```
Comment (yuvalbenshalom) [inline Integrations/FeedTAXII/FeedTAXII.py]: general comment: all functions and cases need a descriptive docstring.

File: Integrations/FeedTAXII/FeedTAXII.py [hunk 4]
```diff
@@ -0,0 +1,918 @@
...
+                    continue
+
+                r['stix_file_{}'.format(r2['htype'])] = r2['indicator']
+
+            r.update(bprops)
+
+        return result
+
+
+''' uriobject.py '''
```
Comment (yuvalbenshalom) [inline Integrations/FeedTAXII/FeedTAXII.py]: remove

File: Integrations/FeedTAXII/FeedTAXII.py [hunk 5]
```diff
@@ -0,0 +1,918 @@
...
+                item = tmp[0]
+                http_client_request = item.get('http_client_request', None)
+                if http_client_request is not None:
+                    http_request_header = http_client_request.get('http_request_header', None)
+                    if http_request_header is not None:
+                        raw_header = http_request_header.get('raw_header', None)
+                        if raw_header is not None:
+                            return [{
+                                'indicator': raw_header.split('\n')[0],
+                                'type': 'http-session',  # we don't support this type natively in demisto
```
Comment (yuvalbenshalom) [inline Integrations/FeedTAXII/FeedTAXII.py]: what will happen if we push an indicator that dose not exist?

**Comment by DeanArbel** [comment]
@kirbles19, please see
- [ ] [RN](https://github.com/demisto/content/edit/taxii-feed/Integrations/FeedTAXII/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5268)
- [ ] [YAML](https://github.com/demisto/content/edit/taxii-feed/Integrations/FeedTAXII/FeedTAXII.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F5268)
- [ ] [Detailed description](https://github.com/demisto/content/edit/taxii-feed/Integrations/FeedTAXII/FeedTAXII_description.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5268)

File: Integrations/FeedTAXII/FeedTAXII.py (hunk 5 above)
Comment (DeanArbel) [reply Integrations/FeedTAXII/FeedTAXII.py]: `demisto.createIndicators` will drop it (i.e. not create the indicator)

**Comment by kirbles19** [comment]
@DeanArbel Done

File: Integrations/FeedTAXII/FeedTAXII.py (hunk 5 above)
Comment (DeanArbel) [reply Integrations/FeedTAXII/FeedTAXII.py]: The server will drop it.
Since it wasn't yet decided if we'll support this type or not, I kept it

**Pull request merged**